src/App/Version/TH.hs

```diff
@@ -1,4 +1,10 @@
 {-# LANGUAGE TemplateHaskell #-}
+-- We disable overlapping-patterns in this module to handle the case statement
+-- in `getCurrentTag`. GHC always thinks one of the two cases is redundant
+-- because `tGitInfoCwdTry` is compiled by Template Haskell into a concrete
+-- Right or Left. However, both branches are needed to handle different
+-- compilation environments.
+{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
```

Comment:
Nit: If you just saved the output of `$$(tGitInfoCwdTry)` to a variable you could avoid suppressing this warning, right?  That might at least look less scary.  The fact that you put in a clear comment is enough for me.

Reply:
I tried that, but still got the warning. GHC is just too clever about this optimization.